src/main/java/bean/command/Bean.java

```diff
@@ -0,0 +1,320 @@
+        System.out.println(SEPARATOR_LINE +
+                reply + "\n" +
+                SEPARATOR_LINE);
+    }
+
+    public static void printTasks() throws EmptyListException {
+        if (tasks.isEmpty()) {
+            throw new EmptyListException();
+        }
+
+        System.out.println(SEPARATOR_LINE +
+                INDENT + "TO DO LIST:\n");
+
+        for (int i = 0; i < tasks.size(); i++) {
+            System.out.println(INDENT + (i + 1) + ". " + tasks.get(i).toString());
+        }
+        System.out.println(SEPARATOR_LINE);
+    }
+
+    // Extract task number as int from user input for mark, unmark and delete commands
+    public static int obtainTaskNum(String userInput) throws TaskNumOutOfBoundsException {
+        // Obtain task number by taking second word of input and trim any spaces, then parse as int
+        String[] words = userInput.split(" ");
+        int taskNum = Integer.parseInt(words[1].trim());
+        if (taskNum < 0 || taskNum > tasks.size()) {
+            throw new TaskNumOutOfBoundsException();
+        }
+        return taskNum;
+    }
+
+    public static void markTaskAsDone(int taskNum) throws IOException {
+
+        int taskIndex = taskNum - 1;
+        tasks.get(taskIndex).setStatus(true);
+        // Confirmation message
+        printFormattedReply(INDENT + "Task " + taskNum + " has been marked as DONE:\n" +
+                INDENT + INDENT + tasks.get(taskIndex).toString());
+        overwriteDataFile();
+    }
+
+    public static void unmarkTaskAsDone(int taskNum) throws IOException {
+        int taskIndex = taskNum - 1;
+        tasks.get(taskIndex).setStatus(false);
+        // Confirmation message
+        printFormattedReply(INDENT + "Task " + taskNum + " has been marked as UNDONE:\n" +
+                INDENT + INDENT + tasks.get(taskIndex).toString());
+        overwriteDataFile();
+    }
+
+    public static void addTask(String userInput, TaskType taskType) throws InsufficientSpaceException, IOException {
+        if (tasks.size() >= MAX_LIST_COUNT) {
+            throw new InsufficientSpaceException();
+        }
+
+        if (taskType == TaskType.TODO) {
+            // Extract description
+            String description = userInput.split("todo ")[1].trim();
+
+            tasks.add(new Todo(description));
+
+        } else if (taskType == TaskType.DEADLINE) {
+            // Extract description and by
+            String[] parts = userInput.split("/by ");
+            // parts: [0] = "deadline {description} ", [1] = " {by}"
+            String description = parts[0].substring("deadline ".length()).trim();
+            String by = parts[1].trim();
+
+            tasks.add(new Deadline(description, by));
+
+        } else {
+            // taskType == TaskType.EVENT
+            // Extract description, from and to
+            String[] splitDescription = userInput.split("/from");
+            // splitDescription: [0] = "event {description} ", [1] = "{from} /to {to}"
```

Comment:
For comments: try focusing more on what the code does and why it's important, rather than how it works. This helps clarify the purpose for future readers.